tools/debugger/debugger_core.py

```diff
@@ -511,6 +520,24 @@ def set_colors(self):
         self.yellow_bg = '\033[43m\033[30m'
         self.blue = '\033[94m'
 
+    def store_client_sources(self, args):
+        self.client_sources = args
+
+    def send_client_source(self):
+        if not self.client_sources:
+            self.send_there_was_no_source()
+            return
+        path = self.client_sources[0]
```

Comment:
Is this an array? If we only use the 0th item, it could simply be a string.
If we want to support multiple sources (could be later), we should make --debugger-wait-source is shell.cpp a loop which waits until an empty string is received.